File: 387-first-unique-character-in-a-string/387-first-unique-character-in-a-string.c
int firstUniqChar(char * s){
    int count[26] = {0};
    int index[26] = {-1};
        
    for(int i = 0; s[i]; i++){
        count[s[i] - 'a']++;
        index[s[i] - 'a'] = i;
    }
    
    for(int i = 0; s[i]; i++){
        if(count[s[i] - 'a'] == 1)
            return index[s[i] - 'a'];
    }
    return -1;
}

File: 387-first-unique-character-in-a-string/387-first-unique-character-in-a-string.java
class Solution {
    public int firstUniqChar(String s) {
        int[] hash = new int[26];
        
        for(int i = 0; i < s.length(); i++)
            hash[s.charAt(i) - 'a']++;
        
        for(int i = 0; i < s.length(); i++){
            if(hash[s.charAt(i) - 'a'] == 1)
                return i;
        }
        return -1;
    }
}